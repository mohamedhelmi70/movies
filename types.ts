export type RootParamList = {
    Movies: undefined;
    Movie: {
        movieId: number;
        movie: Movie;
    };
};

export interface Movie {
    poster_path: string;
    adult?: boolean;
    overview: string;
    release_date: string,
    genre_ids: number[],
    id: number,
    original_title: string,
    original_language?: string,
    title: string,
    backdrop_path?: string,
    popularity?: number,
    vote_count?: number,
    video?: boolean,
    vote_average: number,
    genres?: Genre[]
}

export interface Genre {
    id: number;
    name: string;
}

export interface Credit {
    adult?: boolean,
    gender?: number,
    id: number,
    known_for_department?: string,
    name: string,
    original_name: string,
    popularity?: number,
    profile_path: string,
    cast_id?: number,
    character?: string,
    credit_id: string,
    order?: number
}