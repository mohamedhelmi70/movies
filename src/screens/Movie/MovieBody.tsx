import React from 'react';
import {StyleSheet} from 'react-native';
import {Movie} from '../../../types';
import {FastImage} from '../../components';
import {Text, View} from '../../components/Themed';
import Colors from '../../constants/Colors';
import {wp} from '../../constants/Layout';

interface MovieBodyProps {
    movie: Movie;
}

function MovieBody({movie}: MovieBodyProps) {
    return (
        <View style={styles.container}>
            {Boolean(movie?.poster_path) && (
                <View style={styles.posterView}>
                    <FastImage
                        url={`https://www.themoviedb.org/t/p/original${movie?.poster_path}`}
                        style={styles.poster}
                    />
                </View>
            )}
            <Text style={styles.title} weight="700" font="Roboto-Bold">
                {movie?.original_title}
            </Text>
            {movie?.vote_average && (
                <View style={{marginTop: wp('2%')}}>
                    <Text style={styles.vote} weight="700" font="Roboto-Medium" lightColor={Colors.light.primary}>
                        {movie?.vote_average * 10}%
                    </Text>
                </View>
            )}
            <View style={styles.overview}>
                <Text weight="700" font="Roboto-Bold" style={styles.subtitle}>
                    Overview
                </Text>
                <Text lightColor="#888" style={styles.paragraph}>
                    {movie?.overview}
                </Text>
            </View>
            {movie?.genres && movie?.genres?.length > 0 && (
                <View style={styles.overview}>
                    <Text weight="700" font="Roboto-Bold" style={styles.subtitle}>
                        Genres
                    </Text>
                    <View style={styles.genresView}>
                        {movie?.genres?.map((genre) => (
                            <View key={genre?.id} style={styles.genre} lightColor="#ccc">
                                <Text style={{fontSize: wp('3%')}} weight="400" font="Roboto-Light">
                                    {genre?.name}
                                </Text>
                            </View>
                        ))}
                    </View>
                </View>
            )}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
    },
    posterView: {
        marginBottom: wp('5%'),
    },
    poster: {
        width: wp('30%'),
        height: wp('60%'),
        borderRadius: wp('3%'),
    },
    title: {
        fontSize: wp('4.8%'),
        maxWidth: wp('80%'),
    },
    vote: {
        fontSize: wp('5%'),
    },
    overview: {
        width: wp('90%'),
        marginTop: wp('5.5%'),
    },
    paragraph: {
        marginTop: wp('2%'),
    },
    subtitle: {
        fontSize: wp('3.9%'),
    },
    genresView: {
        marginTop: wp('3%'),
        maxWidth: wp('90%'),
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'flex-start',
        flexWrap: 'wrap',
    },
    genre: {
        borderRadius: wp('2%'),
        paddingHorizontal: wp('1.5%'),
        paddingVertical: wp('.5%'),
        marginBottom: wp('1%'),
        marginRight: wp('1%'),
    },
});

export default MovieBody;
