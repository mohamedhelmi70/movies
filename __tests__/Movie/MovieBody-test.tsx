import React from 'react';
import renderer from 'react-test-renderer';
import {Movie} from '../../types';
import MovieBody from '../../src/screens/Movie/MovieBody';

test('renders MovieBody correctly', () => {
    const movieItem: Movie = {
        id: 123,
        original_title: 'test',
        poster_path: '/poster.png',
        vote_average: 3,
        title: 'test',
        genre_ids: [1, 2],
        release_date: '1994-09-10',
        overview: 'overview test',
    };
    const tree = renderer.create(<MovieBody movie={movieItem} />).toJSON();
    expect(tree).toMatchSnapshot();
});
