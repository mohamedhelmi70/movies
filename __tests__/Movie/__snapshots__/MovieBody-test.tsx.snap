// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieBody correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        Object {
          "marginBottom": 37.5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 22.5,
            "height": 450,
            "width": 225,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "cache": "immutable",
            "headers": Object {
              "Authorization": "4f298a53e552283bee957836a529baec",
            },
            "priority": "normal",
            "uri": "https://www.themoviedb.org/t/p/original/poster.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Roboto",
          "fontWeight": "700",
          "textAlign": "left",
        },
        Object {
          "fontSize": 36,
          "maxWidth": 600,
        },
      ]
    }
  >
    test
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        Object {
          "marginTop": 15,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#79c700",
            "fontFamily": "Roboto",
            "fontWeight": "700",
            "textAlign": "left",
          },
          Object {
            "fontSize": 37.5,
          },
        ]
      }
    >
      30
      %
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        Object {
          "marginTop": 41.5,
          "width": 675,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto",
            "fontWeight": "700",
            "textAlign": "left",
          },
          Object {
            "fontSize": 29.5,
          },
        ]
      }
    >
      Overview
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#888",
            "fontFamily": "Roboto",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Object {
            "marginTop": 15,
          },
        ]
      }
    >
      overview test
    </Text>
  </View>
</View>
`;
